libyul/backends/evm/EVMDialect.cpp

```diff
@@ -161,11 +161,34 @@ map<YulString, BuiltinFunctionForEVM> createBuiltins(langutil::EVMVersion _evmVe
 				_assembly.appendConstant(0);
 			else
 			{
-				yulAssert(
-					_context.subIDs.count(dataName) != 0,
-					"Could not find assembly object <" + dataName.str() + ">."
-				);
-				_assembly.appendDataOffset(_context.subIDs.at(dataName));
+				AbstractAssembly::SubID currentSubId;
+				if (_context.subIDs.count(dataName) == 0)
+				{
+
+					size_t previousDotPos = 0;
+					size_t currentDotPos = dataName.str().find_first_of('.');
+					while (currentDotPos != string::npos)
+					{
+						std::string currentObjectName = dataName.str().substr(previousDotPos, currentDotPos - previousDotPos);
+						yulAssert(
+							_context.subIDs.count(YulString{currentObjectName}) ||
+								_context.currentObject->name == YulString{currentObjectName},
+							"Could not find assembly object <" + currentObjectName + ">."
+						);
+						previousDotPos = currentDotPos + 1;
+						currentDotPos = dataName.str().find_first_of('.', previousDotPos);
+					}
+					std::string currentObjectName = dataName.str().substr(previousDotPos);
+					yulAssert(
+						_context.subIDs.count(YulString{currentObjectName}),
+						"Could not find assembly object <" + currentObjectName + ">."
+					);
+					currentSubId = _context.subIDs.at(YulString{currentObjectName});
+				}
+				else
+					currentSubId = _context.subIDs.at(dataName);
+
+				_assembly.appendDataOffset(currentSubId);
```

Comment:
This can also be simplified if we check that it is in a form of <contectObjectName>.<subObjectName>
and assert on _context.subIDs.count(<subObjectName>)